pkg/controller/controller.go

```diff
@@ -228,7 +228,29 @@ func sendEvent(obj interface{}, c *config.Config, kind string, eventType config.
 	// Send notification to communication channel
 	if c.Communications.Slack.Enabled {
 		notifier = notify.NewSlack(c)
-		go notifier.SendEvent(event)
+
+		// empty value in event.channel sends notifications to default channel.
+		if event.Channel == "" {
+			go notifier.SendEvent(event)
+		} else {
+			// non empty value in event.channel demands redirection of events to a different channel
+			// clone config and update Slack.Channel
+			newC := *c
```

Comment:
I believe this should be handled in `notifier.SendEvent`, where `event.Channel` if set we send to channel from event.
Logic should be more cleaner in this case. notifier will give preference to `event.Channel` over configured channel.
This also gives flexibility for notifier to use Channel as it wants , across different type of notifiers.

Reply:
resolved it.